usb.c

```diff
@@ -1253,8 +1258,20 @@ int usb_context_scan(struct iio_scan_result *scan_result, const char *args)
 		return ret;
 
 	ret = libusb_init(&ctx);
-	if (ret < 0)
+	if (ret < 0) {
+#ifdef __linux__
+		/* When Linux's OTG USB is in device mode, and there are no hosts,
+		 * libusb_init() is expected to fail, but we shouldn't treat that
+		 * as a hard failure - only that there are no devices.
+		 */
+		DIR* dir = opendir("/dev/bus/usb/");
+		if (dir)
+			closedir(dir);
+		else if (errno == ENOENT)
+			ret = 0;
```

Comment:
You could `return 0` here, then you don't need to add a `LIBUSB_SUCCESS` (which shouldn't really be an "error" code IMHO).

Reply:
Updated per your request...